File: qe-selenium/qe-selenium-python/Activity12.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.common.action_chains import ActionChains
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC

with webdriver.Firefox() as driver:
    driver.get("https://training-support.net/webelements/dynamic-content/")
    wait=WebDriverWait(driver,10)
    print(driver.title)
    button=driver.find_element(By.XPATH,"//*[@id=\"genButton\"]")
    button.click()
    content=driver.find_element(By.XPATH,"//*[@id=\"word\"]")
    wait.until(EC.text_to_be_present_in_element((By.ID,'word'),"soccer"))
    driver.quit

File: qe-selenium/qe-selenium-java/src/main/java/com/demos/Activity12.java
package com.demos;

import java.time.Duration;

import org.openqa.selenium.By;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;

public class Activity12 {
    public static void main(String[] args) {
        FirefoxDriver driver=new FirefoxDriver();
        driver.get("https://training-support.net/webelements/dynamic-content/");
        driver.getTitle();
        driver.findElement(By.xpath("//*[@id=\"genButton\"]")).click();
        WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(10));
        wait.until(ExpectedConditions.textToBePresentInElementLocated((By.xpath("//*[@id=\"word\"]")),"release"));
        System.out.println(driver.findElement(By.xpath("//*[@id=\"word\"]")).getText());
        driver.quit();

       
    }
    
}
